src/main.c

```diff
@@ -2864,6 +2864,16 @@ int main(int argc, char** argv)
 		fprintf(stderr,  "failed to initialize list addresses\n");
 		goto error;
 	}
+	for(struct socket_info* si=udp_listen; si; si=si->next){
+		udp_set_src_addr(&si->address);
+	}
```

Comment:
This loop results is setting the global variables to the last IPv4 or IPv6 listen addresses in the list, right?

Same happens with the loop for tcp below.

Reply:
Yes, correct. The loop result is setting the global variables to the last IPv4 or IPv6 listen addresses in the list. The same for UDP.
But these loops are not correct since these source IP addresses are set up by config file (at least for TCP), these loops just replace the source IP addresses. I will change it.